### PR TITLE
Translation capability for branch 0_15

### PR DESCRIPTION
**Pull request by Sojourneer**
Translation capability for WLED. Google translated language files provided for JA, FR, DE.  Has been tested on an ESP8266.

The coverage is high but not perfect; manual addition of text is required.  A more robust solution using a screen scraper on a running WLED instance is probably a better way to go. 

From the introduction to the I18N/tools (see wled00/I18N/tools/I18N.md for the details): 

# WLED Localization

Localization for WLED provides the ability to switch between display languages dynamically from the UI settings page.  This setting is local to the browser, and is effective immediately (no Save or page refresh is required).

The rest of this document explains the process for providing this capability. 

In summary, the UISimulator Web Server and a couple of python scripts are used to extract, translate, and test the translations on a static simulation of WLED.  The translation files are then committed to Github, from which the L12N script included on each HTML page will access and apply the translation.
****

## Release Notes

* **New Features**
  * Added multi-language support enabling users to switch the UI display language to French, German, or Japanese.
  * Added a language selection dropdown in the UI Settings page with automatic language preference persistence.

* **Chores**
  * Updated build system to include language resources and localization assets.

### CODE REVIEW

**Comment by softhack007** [comment]
@Sojourneer please clen up you PR - its seems there are lots of copied files and "cached whatever" that look like backups or intermediate files.

I'll mark your PR as draft for the time being.

File: wled00/UISimulator/runSimulator [hunk 1]
```diff
@@ -0,0 +1,2 @@
+cd wled00
+sudo python3 UISimulator/main.py "$@"
```
Comment (softhack007) [inline wled00/UISimulator/runSimulator:2]: `sudo` should not be needed in a public repository.

**Comment by softhack007** [comment]
> The rest of this document explains the process for providing this capability.

Where is this document?

File: wled00/UISimulator/runSimulator (hunk 1 above)
Comment (Sojourneer) [reply wled00/UISimulator/runSimulator:2]: I don't know what that means.  For example, is it permissible in documentation in a public repository?

**Comment by Sojourneer** [comment]
> @Sojourneer please clean up you PR - it seems there are lots of copied files and "cached _whatever_" that look like backups or intermediate files.
> 
> I'll mark your PR as draft for the time being.

There are no files that are purely generated from something else.

**Comment by Sojourneer** [comment]
> From the introduction to the I18N/tools (see wled00/I18N/tools/I18N.md for the details):

**Comment by softhack007** [comment]
> > I'll mark your PR as draft for the time being.
> 
> There are no files that are purely generated from something else.

Maybe i should be a bit clearer about what I want.

A/ New directories you added --> please move `I18N`, `UIsimulator`, and `scripts` out of `wled00`. What will be included into the firmware itself, should be subfolders inside `data`. `UIsimulator`, and `scripts` could be placed into "tools"


image


B/ There are many sub-folders inside I18N/data and their purpose is unclear - in fact these looks like cached files. Please check if they can be removed and auro-generated by script. As a rule of thumb, any file that can be regenerated by a build script should _not_ be added to the github repository. 

image


After this first cleanup, other maintainers might be able to review your work, ask questions and suggest improvements.

File: wled00/UISimulator/runSimulator (hunk 1 above)
Comment (softhack007) [reply wled00/UISimulator/runSimulator:2]: it means that i don't want to have scripts that use `sudo`. 
Running programs with root permissions is only permissible when there is a very good justification.

**Comment by Sojourneer** [comment]
> > > I'll mark your PR as draft for the time being.
> > 
> > 
> > There are no files that are purely generated from something else.
> 
> Maybe i should be a bit clearer about what I want.
> 
> A/ New directories you added --> please move `I18N`, `UIsimulator`, and `scripts` out of `wled00`. What will be included into the firmware itself, should be subfolders inside `data`. `UIsimulator`, and `scripts` could be placed into "tools"
> 
> After this first cleanup, other maintainers might be able to review your work, ask questions and suggest improvements.

Sorry, I missed your message.

I decided a few weeks back to move all the work to a new repository https://github.com/Sojourneer/WLED-Translators-Pack, so all the ugly stuff is elsewhere and it is somewhat independent of the main repository.  The Translator's Pack also includes a proxy server instead of the UI Simulator,which will give better coverage of the application.  The proxy server stuff has me tearing my hair out, so I've been a bit delayed.

I will get back to it after New Years.

**Comment by softhack007** [comment]
closing due to inactivity - the PR structure (files, scripts, unclear helpers, ...) needs heavy cleanup before it would be acceptable for merging.

**Comment by softhack007** [comment]
@coderabbitAI review
